src/scheduler.rs

```diff
@@ -69,27 +99,28 @@ pub fn generate_ids(
     instance: &str,
     eligible_nodes: &EligibleNodesForRoleAndGroup,
 ) -> Vec<PodIdentity> {
-    let mut id: u16 = 1;
     let mut generated_ids = vec![];
     for (role_name, groups) in eligible_nodes {
         for (group_name, eligible_nodes) in groups {
             let ids_per_group = eligible_nodes
                 .replicas
                 .map(usize::from)
                 .unwrap_or_else(|| eligible_nodes.nodes.len());
-            for _ in 1..ids_per_group + 1 {
+            for id in 1..ids_per_group + 1 {
                 generated_ids.push(PodIdentity {
                     app: app_name.to_string(),
                     instance: instance.to_string(),
                     role: role_name.clone(),
                     group: group_name.clone(),
                     id: id.to_string(),
                 });
-                id += 1;
             }
         }
     }
 
+    // Sort the result to make testing predictable. Otherwise the for-loop above is not
+    // guaranteed to preserve insertion order so the tests might fail at random.
+    generated_ids.sort_by(|a, b| a.partial_cmp(b).unwrap());
```

Comment:
`generated_ids.sort()` doesnt do the trick?

Reply:
No because `sort()` is in-place and doesn't return anything.